### DISCUSSION TITLE
UI Redo

### DISCUSSION THREAD

**Discussion by nckrwlmn** [blakeblackshear/frigate#15710] [Ask A Question]
Is frigate going to be receiving a UI upgrade where there we can do all things through the UI, without having to deal with any config files? Pretty please? Just thought I'd ask.

**Comment by hawkeye217** [comment]
This is a very large undertaking that will likely be rolled out over many coming versions. There is no timeline or roadmap, however.

**Comment by bartbutenaers** [comment]
Hi, I started with Frigate today and I am very impressed.  But I also have to admit that the yaml config is a bit hard to get started with.  Would be very helpful if at least the ***required*** settings could become configurable via the ui.  But I understand that there is a lot of other work todo by a small team...

**Reply by hawkeye217** [reply]
Yes, there are only two of us that do most of the work on the UI, and we are volunteers for the project. This will come - eventually.

**Reply by bartbutenaers** [reply]
Yes I completely understand.  Since we have a similar problem in the Node-RED community, I first had a look at the contributor log of this repo.  And it was immediately very clear to me how many people are in charge here...

I am already drawning in open-source work, so cannot much be a help here unfortunately 😞 
But I know somebody that is going to start using Home Assistant, and he certainly has the necessary skills/mindset to contribute such a feature.  I will ask him after my holidays if he has time/interest to support this project by adding such a feature to the ui.  Please ***don't*** put high hopes on it, in case it won't happen.  But at least I can ask him...

It would be useful if you could give me some pointers, so I can at least give him a rather good description of what is needed to allow the ***required*** settings from the config.yaml to be managed via the web ui.  Of course I want to _**keep things simple**_, otherwise I am pretty sure I won't be able to convince him...

Some questions that already popup in my head:
1. Am I correct that a camera should be added and maintained in the ui, similar to the camera groups (i.e. via an extra icon in the left menu bar)?

   ![image](https://github.com/user-attachments/assets/129272b1-5c6f-4a45-825f-2d93744e0735)

2. From the [reference](https://docs.frigate.video/configuration/reference/) documentation, the following settings should be configurable for each camera:
   ```
   # Required
   cameras:
     # Required: name of the camera
     name_of_the_camera:
       # Optional: Enable/Disable the camera (default: shown below).
       # If disabled: config is used but no live stream and no capture etc.
       # Events/Recordings are still viewable.
       enabled: True
       # Required: ffmpeg settings for the camera
       ffmpeg:
         # Required: A list of input streams for the camera. See documentation for more information.
         inputs:
           # Required: the path to the stream
           # NOTE: path may include environment variables or docker secrets, which must begin with 'FRIGATE_' and be referenced in {}
           - path: rtsp://viewer:{FRIGATE_RTSP_PASSWORD}@10.0.10.10:554/cam/realmonitor?channel=1&subtype=2
             # Required: list of roles for this stream. valid values are: audio,detect,record
             # NOTICE: In addition to assigning the audio, detect, and record roles
             # they must also be enabled in the camera config.
             roles:
               - audio
               - detect
               - record
   ```
   However I saw an example file where they put - beneath the camera name -  extra stuff:
   ```
    detect:
      enabled: true # <---- disable detection until you have a working camera feed
      fps: 3
    record:
      enabled: true
      events:
        pre_capture: 10
        post_capture: 10
        retain:
          default: 21
   ```
   But not sure if that all fits on a single popup dialog.  Not sure if you guys already had some idea in mind?

Thanks!

**Reply by NickM-27** [reply]
I think a lot more discussion would be needed, and in general we encourage anyone who plans to contribute a large change to create a discussion outlining their idea in detail first to ensure it matches what we are planning. 

In this case, this would absolutely need to be split into multiple pieces and in many cases isn't simply just a lack of UI but also requires backend changes.
